lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -893,25 +896,147 @@ def default_ami_for_region(region, criteria = {})
+        end
+        # decrypt so we can utilize
+        private_key = OpenSSL::PKey::RSA.new(get_private_key(instance.key_name))
+        password = private_key.private_decrypt decoded
+      end
 
-      decoded = Base64.decode64(encrypted_admin_password)
-      private_key = OpenSSL::PKey::RSA.new(pem_bytes)
-      decrypted_password = private_key.private_decrypt decoded
+      disable_sspi =  machine_options[:winrm_disable_sspi] || false # default to Negotiate
+      basic_auth_only = machine_options[:winrm_basic_auth_only] || false # disallow Basic auth by default
+      no_ssl_peer_verification = machine_options[:winrm_no_ssl_peer_verification] || false #disallow MITM potential by default
 
       winrm_options = {
-        :user => machine_spec.reference['winrm_username'] || 'Administrator',
-        :pass => decrypted_password,
-        :disable_sspi => true,
-        :basic_auth_only => true
+        user: username,
+        pass: password,
+        disable_sspi: disable_sspi,
+        basic_auth_only: basic_auth_only,
+        no_ssl_peer_verification: no_ssl_peer_verification,
       }
 
+      if no_ssl_peer_verification or type != :ssl
+        # =>  we won't verify certs
+        pass
+      elsif machine_spec.reference[:winrm_ssl_thumbprint]
+        #      elsif machine_spec.reference[:winrm_ssl_cert]
+        # we have stored the cert
+        pass
+      else
+        # we need to retrieve the cert and verify it by connecting just to
+        # retrieve the ssl certificate and compare it to what we see in the
+        # console logs
+        console_lines = Base64.decode64(instance.console_output.data.output).lines
+        noverify_peer_context = OpenSSL::SSL::SSLContext.new
+        noverify_peer_context.verify_mode = OpenSSL::SSL::VERIFY_NONE
+        tcp_connection = TCPSocket.new(instance.private_ip_address, '5986')
+        shady_ssl_connection = OpenSSL::SSL::SSLSocket.new(tcp_connection, noverify_peer_context)
+        shady_ssl_connection.connect
+        shady_ssl_connection.peer_cert_chain
+        winrm_cert = shady_ssl_connection.peer_cert_chain.first
+
+        rdp_thumbprint = console_lines.grep(
+          /RDPCERTIFICATE-THUMBPRINT/)[-1].split(': ').last.chomp
+        rdp_subject = console_lines.grep(
+          /RDPCERTIFICATE-SUBJECTNAME/)[-1].split(': ').last.chomp
+        winrm_subject = winrm_cert.subject.to_s.split('=').last.upcase
+        winrm_thumbprint=OpenSSL::Digest::SHA1.new(winrm_cert.to_der).to_s.upcase
+
+        if rdp_subject != winrm_subject or rdp_thumbprint != winrm_thumbprint
+          Chef::Log.fatal "Winrm ssl port certificate differs from rdp console logs"
+        end
+        # if machine_spec.reference[:winrm_ssl_subject] != winrm_subject
+        #   machine_spec.reference[:winrm_ssl_subject] = winrm_subject
+        # end
+        if machine_spec.reference[:winrm_ssl_thumbprint] != winrm_thumbprint
+          machine_spec.reference[:winrm_ssl_thumbprint] = winrm_thumbprint
```

Comment:
Storing verified winrm_thumbprint for later use, we could also store certificate, but not really that useful since we won't be relying on it.  See WinRb/WinRM#151